### PR TITLE
#Task 1

### PR DESCRIPTION
**Pull request by MEHV29**
This contains the 3 points of the task.

### CODE REVIEW

File: Program.cs [hunk 1]
```diff
@@ -1,10 +1,181 @@
-﻿namespace Task1
+﻿using System;
+
+namespace Task1
 {
     internal class Program
     {
+        static void DisplayArray(int size, int[] array)
+        {
```
Comment (IgorKheidorov) [inline Program.cs:8]: Use array.Length for array size

File: Program.cs [hunk 2]
```diff
@@ -1,10 +1,181 @@
...
+            for (int i = 0; i < size; i++)
+            {
+                if (i < (size - 1))
+                {
+                    Console.Write($"{array[i]},");
+                }
+                else
+                {
```
Comment (IgorKheidorov) [inline Program.cs]: Unnecessary comparison    i < (size - 1)), just  add Console.Write($"\n"); after loop for new line

File: Program.cs [hunk 3]
```diff
@@ -1,10 +1,181 @@
...
+                    Console.Write($"{array[i]}]\n");
+                }
+            }
+        }
+
+        static bool ConvertToDuodecimal(int decimalNumber)
+        {
+            string duodecimal = "";
+            char duodecimalDigit;
+
```
Comment (IgorKheidorov) [inline Program.cs:27]: What about negative values? you have to process them also

File: Program.cs [hunk 4]
```diff
@@ -1,10 +1,181 @@
...
+                {
+                    Console.Write($"{array[i]},");
+                }
+                else
+                {
+                    Console.Write($"{array[i]}]\n");
+                }
+            }
+        }
+
```
Comment (IgorKheidorov) [inline Program.cs:16]: Not good name for the method, you do not convert decimal to duodecimal here, you just check for AA in each number. Change the name

File: Program.cs [hunk 5]
```diff
@@ -1,10 +1,181 @@
...
+                {
+                    duodecimalDigit = (char)(remainder + '0');
+                }
+                else
+                {
+                    duodecimalDigit = (char)(remainder - 10 + 'A');
+                }
+
+                duodecimal = duodecimalDigit + duodecimal;
+
```
Comment (IgorKheidorov) [inline Program.cs:42]: Usage of string methods (for example, Contains) is not allowed now )) Only in future

File: Program.cs [hunk 6]
```diff
@@ -1,10 +1,181 @@
...
+                if(duodecimal.Contains("AA"))
+                {
```
Comment (IgorKheidorov) [inline Program.cs:49]: You break the loop after "AA" is found, but you need to find numbers with *exact two A (not less or more)

File: Program.cs [hunk 7]
```diff
@@ -1,10 +1,181 @@
...
-            Console.WriteLine("Hello, World!");
+            //Task 1.1
+            Console.WriteLine("Please enter number a:");
+            string inputA = Console.ReadLine();
+            Console.WriteLine("Please enter number b:");
+            string inputB = Console.ReadLine();
+
+            int numberA = int.Parse(inputA);
+            int numberB = int.Parse(inputB);
+
```
Comment (IgorKheidorov) [inline Program.cs:67]: Unclear, you have to add comment why 130

File: Program.cs [hunk 8]
```diff
@@ -1,10 +1,181 @@
...
+        }
+
         static void Main(string[] args)
         {
-            Console.WriteLine("Hello, World!");
+            //Task 1.1
+            Console.WriteLine("Please enter number a:");
+            string inputA = Console.ReadLine();
+            Console.WriteLine("Please enter number b:");
+            string inputB = Console.ReadLine();
```
Comment (IgorKheidorov) [inline Program.cs:63]: What about if A > B ? process this case

File: Program.cs [hunk 9]
```diff
@@ -1,10 +1,181 @@
...
+                int digit = int.Parse(isbn[i].ToString());
+
+                sumProducts += digit * weightPosition;
+
+                weightPosition--;
+            }
+
+            int checkDigit = 11 - (sumProducts % 11);
+            string formattedCheckDigit;
+
```
Comment (IgorKheidorov) [inline Program.cs]: better use ternary operator:  formattedCheckDigit = checkDigit== 10? "X": checkDigit.ToString();

File: Program.cs [hunk 10]
```diff
@@ -1,10 +1,181 @@
...
+            if (checkDigit == 10)
+            {
+                formattedCheckDigit = "X";
+            }
+            else
+            {
+                formattedCheckDigit = checkDigit.ToString();
+            }
+
```
Comment (IgorKheidorov) [inline Program.cs:115]: According to the task such formatting for ISBN not needed, but OK

File: Program.cs [hunk 11]
```diff
@@ -1,10 +1,181 @@
...
+                {
+                    if (originalArray[i] == tempArray[j])
+                    {
+                        isDuplicate = true;
+                        break;
+                    }
+                }
+
+                if (!isDuplicate)
+                {
```
Comment (IgorKheidorov) [inline Program.cs:165]: better  tempArray[uniqueCount++] = originalArray[i];, and line 164 is not needed

**Review by IgorKheidorov — CHANGES_REQUESTED** [review]
Cool in total, just think about simplifying code in some cases, fix logics in line 42

**Comment by MEHV29** [comment]
I have done the changes, please, let me know anything, thanks!

File: Program.cs (hunk 1 above)
Comment (MEHV29) [reply Program.cs:8]: Modified

File: Program.cs (hunk 2 above)
Comment (MEHV29) [reply Program.cs]: Modified

File: Program.cs (hunk 3 above)
Comment (MEHV29) [reply Program.cs:27]: Fixed

File: Program.cs (hunk 4 above)
Comment (MEHV29) [reply Program.cs:16]: Modified

File: Program.cs (hunk 5 above)
Comment (MEHV29) [reply Program.cs:42]: Modified

File: Program.cs (hunk 6 above)
Comment (MEHV29) [reply Program.cs:49]: Fixed

File: Program.cs (hunk 7 above)
Comment (MEHV29) [reply Program.cs:67]: Modified

File: Program.cs (hunk 8 above)
Comment (MEHV29) [reply Program.cs:63]: Fixed

File: Program.cs (hunk 9 above)
Comment (MEHV29) [reply Program.cs]: Modified

File: Program.cs (hunk 11 above)
Comment (MEHV29) [reply Program.cs:165]: Modified

File: Program.cs [hunk 12]
```diff
@@ -1,10 +1,190 @@
...
+            }
+
+            for(int i = 0; i<duodecimal.Length; i++)
+            {
+                if (duodecimal[i] == 'A')
+                {
+                    counterAA++;
+                }
+            }
+
```
Comment (IgorKheidorov) [inline Program.cs:53]: More compact:          return counterAA == 2;

**Review by IgorKheidorov — APPROVED** [review]
Cool. Fix line 54 and can merge )))

File: Program.cs (hunk 12 above)
Comment (MEHV29) [reply Program.cs:53]: Modified

**Pull request merged**